src/main/java/org/candlepin/subscriptions/files/ProductIdToProductsMapSource.java

```diff
@@ -23,16 +23,13 @@
 import org.candlepin.subscriptions.ApplicationProperties;
 import org.candlepin.subscriptions.util.ApplicationClock;
 
-import org.springframework.stereotype.Component;
-
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
 /**
  * Loads the product ID to list of Tally products mapping from a YAML file.
  */
-@Component
```

Comment:
This raises an interesting question.  I see you made a bean for this class in `ProductIdMappingConfiguration.java`.  I am assuming you did that to control the creation of the bean via an `@Import` whereas with `@Component` it would be picked up universally.

What if we used [`@Conditional`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Conditional.html) and its cousins instead?  Which conditional we would use would remain to be seen, but we can use any SpEL expression or property or bean.  Just as a thought, the new `Configuration` classes that define a profile's beans could create a marker bean that the `@ConditionalOnBean` annotations could reference.  This would allow us to keep using `@Component` annotations and would cut down on the proliferation of constructing beans by hand.

Reply:
According to Javadocs that you linked, intended use of `@Conditional` is for `Configuration` classes and `@Bean` methods only, so unfortunately, that wouldn't be supported on the Component itself.

One option is to scrap the Configuration-based approach and use `@Profile` all over the place (for all components), but I personally think that makes it much harder to grok what beans are being included in a given profile (and it's difficult to enforce if we kept package scanning enabled globally for all packages).

The approach I favor is to selectively enable component scanning on specific packages. So I will push an update that uses scanning on the files package for the `ProductIdMappingConfiguration`. (This product ID map is a little odd in that it's shared between components in unlike all other file beans). Please let me know if this works for you.

Reply:
Yeah, that is fine by me.  I also realized after I wrote this that having a `@Conditional` and a marker bean is actually *less* succinct and direct than using `@Profile`.  I do like using `@Component` and I think that should drop in nicely with your selective package scanning approach.  Otherwise, it's not like it's really that much of a burden to just write a simple `@Bean` method.  I can just deal with it. :)